crates/routes/src/images.rs

```diff
@@ -123,6 +128,19 @@ async fn full_res(
   client: web::Data<ClientWithMiddleware>,
   context: web::Data<LemmyContext>,
 ) -> Result<HttpResponse, Error> {
+  // block access to images if instance is private and unauthorized, public
+  let site = blocking(context.pool(), Site::read_local_site).await?;
+  // The site might not be set up yet
+  if let Ok(site) = site {
+    if site.private_instance {
+      let jwt = req
+      .cookie("jwt")
+      .expect("No auth header for picture access");
```

Comment:
No expect, do a `.map_err(error::ErrorUnauthorized)`

Reply:
OK, that should be done upload() on line [92](https://github.com/LemmyNet/lemmy/blob/0bec040c4e8d6822755900689af8f2359f6a93ab/crates/routes/src/images.rs#L97) as well i think?